plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -1900,13 +1896,12 @@ void read_write::push_transaction(const read_write::push_transaction_params& par
    try {
       auto pretty_input = std::make_shared<packed_transaction>();
       auto resolver = make_resolver(this, abi_serializer_max_time);
-      transaction_metadata_ptr ptrx;
       try {
          abi_serializer::from_variant(params, *pretty_input, resolver, abi_serializer_max_time);
-         ptrx = std::make_shared<transaction_metadata>( pretty_input, db.configured_subjective_signature_length_limit() );
```

Comment:
Before the `check_variable_sig_size` check was done here and it would throw the `packed_transaction_type_exception` exception. Now that check is delayed until key recovery and I'm not sure how the exception will be handled.

Same goes for `send_transaction`.

It isn't clear whether this is a problem. It depends on what the point of that validation was and when it is expected to be done.

@spoonincode Can I get your input on this?

Reply:
Not `packed_transaction_type_exception`, but rather `sig_variable_size_limit_exception`. It's a subjective check, and a subjective exception. I haven't studied this change enough to know the ramifications for moving this (does it keep the subjectivity?)

Reply:
It is still subjective, but it is now done in `producer_plugin::on_incoming_transaction_async` where the transaction_metadata is created via a future.

Reply:
Yeah I think it's okay -- the check still occurs before the actual recovery starts, so the purpose of the check is still clutched in